Source/Core/ArbitraryProjectionTilingScheme.js

```diff
@@ -0,0 +1,276 @@
+     * @param {Number} level The tile level-of-detail.  Zero is the least detailed.
+     * @param {Object} [result] The instance to which to copy the result, or undefined if a new instance
+     *        should be created.
+     * @returns {Rectangle} The specified 'result', or a new object containing the rectangle
+     *          if 'result' is undefined.
+     */
+    ArbitraryProjectionTilingScheme.prototype.tileXYToRectangle = function(x, y, level, result) {
+        var nativeRectangle = this.tileXYToNativeRectangle(x, y, level, nativeRectangleScratch);
+
+        return Rectangle.approximateCartographicExtents(nativeRectangle, this._projection, result, this._granularity);
+    };
+
+    /**
+     * Calculates the tile x, y coordinates of the tile containing
+     * a given cartographic position.
+     *
+     * @param {Cartographic} position The position.
+     * @param {Number} level The tile level-of-detail.  Zero is the least detailed.
+     * @param {Cartesian2} [result] The instance to which to copy the result, or undefined if a new instance
+     *        should be created.
+     * @returns {Cartesian2} The specified 'result', or a new object containing the tile x, y coordinates
+     *          if 'result' is undefined.
+     */
+    ArbitraryProjectionTilingScheme.prototype.positionToTileXY = function(position, level, result) {
+        if (!Rectangle.contains(this._rectangle, position)) {
+            // outside the bounds of the tiling scheme
+            return undefined;
+        }
+        var projectedPosition = this._projection.project(position);
+
+        return this.nativePositionToTileXY(projectedPosition, level, result);
+    };
+
+    var projectedPositionCartographicScratch = new Cartographic();
+    /**
+     * Calculates the tile x, y coordinates of the tile containing
+     * a given native position.
+     *
+     * @param {Cartesian3} projectedPosition A position in the tiling scheme's projected coordinate system.
+     * @param {Number} level The tile level-of-detail.  Zero is the least detailed.
+     * @param {Cartesian2} [result] The instance to which to copy the result, or undefined if a new instance
+     *        should be created.
+     * @returns {Cartesian2} The specified 'result', or a new object containing the tile x, y coordinates
+     *          if 'result' is undefined.
+     */
+    ArbitraryProjectionTilingScheme.prototype.nativePositionToTileXY = function(projectedPosition, level, result) {
+        var xTiles = this.getNumberOfXTilesAtLevel(level);
+        var yTiles = this.getNumberOfYTilesAtLevel(level);
+        var projectedRectangle = this._projectedRectangle;
+
+        var projectedPositionCartographic = projectedPositionCartographicScratch;
+        projectedPositionCartographic.longitude = projectedPosition.x;
+        projectedPositionCartographic.latitude = projectedPosition.y;
+        //if (!Rectangle.contains(projectedRectangle, projectedPositionCartographic)) {
```

Comment:
Is this not needed because of the clamp below? If so, remove the commented code. Also, `projectedPositionCartographic` wouldn't be needed.